### ISSUE TITLE
Debug option causes dual html/json report generation to fail

### ISSUE THREAD

**Issue by bf4** [presidentbeef/brakeman#922]
My `config/brakeman.yml` had `:debug: true` set, which recently just started causing an exception upon report generation

```
/home/ubuntu/.rvm/gems/ruby-2.2.5@gemset/gems/brakeman-3.3.4/lib/brakeman/report/report_table.rb:25:in
   `generate_report': undefined local variable or method `generate_templates' for #<Brakeman::Report::Table:0x000000023cd5c8> (NameError)
  from /home/ubuntu/.rvm/gems/ruby-2.2.5@gemset/gems/brakeman-3.3.4/lib/brakeman/report.rb:76:in `generate'
  from /home/ubuntu/.rvm/gems/ruby-2.2.5@gemset/gems/brakeman-3.3.4/lib/brakeman/report.rb:67:in `to_s'
  from /home/ubuntu/.rvm/gems/ruby-2.2.5@gemset/gems/brakeman-3.3.4/lib/brakeman/report.rb:38:in `format'
  from /home/ubuntu/.rvm/gems/ruby-2.2.5@gemset/gems/brakeman-3.3.4/lib/brakeman.rb:367:in `block (2 levels) in write_report_to_files'
```

The failure originates in https://github.com/presidentbeef/brakeman/blob/2bd047200fe0ef346b10ac9fee41329e367dd6f9/lib/brakeman/report/report_table.rb

```ruby
if tracker.options[:debug]
 out << "\n+TEMPLATES+\n\n" <<
 truncate_table(generate_templates.to_s) << "\n"
end
```

`generate_templates` is only defined in
https://github.com/presidentbeef/brakeman/blob/0b618fdd1e37683df67844e31e554c347f9aed3c/lib/brakeman/report/report_html.rb but not in
https://github.com/presidentbeef/brakeman/blob/0b618fdd1e37683df67844e31e554c347f9aed3c/lib/brakeman/report/report_json.rb

My brakeman.yml is (I've now set `:debug: false`

```yaml
---
:exit_on_warn: true
:min_confidence: 1
:quiet: false
:run_all_checks: true
:report_routes: true
:message_limit: 200
:table_width: 200
:github_repo: org/repo
:rails3: true
:rails4: true
:ignore_file: config/brakeman.ignore
:debug: false
# :app_path: "."
# :summary_only: true
# :skip_files:
# - config/database.yml
# :safe_methods:
# - :banana
# :url_safe_methods:
# - :banana_url
# :previous_results_json: "$HOME/projects/app/reports/brakeman.json"
# :output_files:
# - reports/brakeman.json
# - reports/brakeman.html
# - "/dev/stdout"
# :comparison_output_file: "/dev/stdout"
```

and I'm running brakeman via the script `bin/brakeman` below.

```
#!/usr/bin/env bash
#
# Usage:
#   bin/brakeman
#
#   If brakeman finds problems, this script will exit with a non-zero exit code.
#   To resolve, dev must run `brakeman -I` and resolve security findings.
# 
# Does not install the latest brakeman if already installed
gem install brakeman --conservative --no-rdoc --no-ri
# Only the output configurations are specified below. The remaining configuration
# is in config/brakeman.yml and any ignored warnings in config/brakeman.ignore
# see https://github.com/presidentbeef/brakeman/blob/master/OPTIONS.md
# config template generated by running:
# brakeman -z -w2 -q -A --routes --message-limit 200 --table-width 200 --github-repo org/repo -4 -i config/brakeman.ignore -d -p .  --summary --skip-files config/database.yml --safe-methods banana  --url-safe-methods banana_url --compare reports/brakeman.json -o /dev/stdout -o reports/brakeman.json -o reports/brakeman.html -C > config/brakeman.yml
# https://github.com/presidentbeef/brakeman/issues/697#issuecomment-129612973
# The input and output files for JSON comparison can be the same.
# The diff is always sent to the first -o option, so in this case printed to the console.
# What this is missing is printing out just the summary.
# It's not currently possible to both summarize and generate full reports.
#
# see
#  https://github.com/presidentbeef/brakeman/issues/697
#  https://medium.com/@sgringwe/rails-security-checks-in-ci-6c366f3535cf
#  https://github.com/thesp0nge/dawnscanner
#  https://twitter.com/presidentbeef/status/591250085314428928
BRAKEMAN_REPORTS_DIR="${BRAKEMAN_REPORTS_DIR:-reports/brakeman}"
mkdir -p "$BRAKEMAN_REPORTS_DIR"
BRAKEMAN_REPORT_BASE="${BRAKEMAN_REPORTS_DIR}/brakeman"
if [ ! -e "{$BRAKEMAN_REPORT_BASE}.json" ]; then
# If no report is present, will show new errors in nice output.
cat << EOF > "${BRAKEMAN_REPORT_BASE}.json"
{
  "scan_info": {
  },
  "warnings": [
  ],
  "ignored_warnings": [
  ],
  "errors": [
  ]
}
EOF
fi
mv "${BRAKEMAN_REPORT_BASE}.json" "${BRAKEMAN_REPORT_BASE}_prev.json"

brakeman -c config/brakeman.yml \
  --compare "${BRAKEMAN_REPORT_BASE}_prev.json" -o /dev/stdout \
  -o "${BRAKEMAN_REPORT_BASE}.json" \
  -o "${BRAKEMAN_REPORT_BASE}.html" \
  -o /dev/stdout
```

**Comment by presidentbeef** [comment]
Can you believe I already fixed this and then forgot to merge it? https://github.com/presidentbeef/brakeman/pull/919

**Comment by presidentbeef** [comment]
Should be fixed in 3.3.5. Thank you for letting me know!

**Comment by bf4** [comment]
:)

B mobile phone
